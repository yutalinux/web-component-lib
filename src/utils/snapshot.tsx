import { type Attributes, type FunctionComponent, createElement } from "react";
import { renderToString } from "react-dom/server";

export function snapshotElement<T extends object>(
  type: FunctionComponent<T>,
  args: (Attributes & T) | null,
) {
  const element = createElement(type, args);
  return renderToString(element);
}
