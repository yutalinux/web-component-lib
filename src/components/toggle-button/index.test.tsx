import { snapshot } from "@/utils/tests/snapshot";
import { expect, test } from "vitest";
import ToggleButton from ".";

test("ToggleButton Snapshot", () => {
  expect(snapshot(ToggleButton, { label: "Toggle Button" })).toMatchSnapshot();
});
