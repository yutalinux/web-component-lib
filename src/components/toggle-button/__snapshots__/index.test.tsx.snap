// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToggleButton Snapshot 1`] = `"<button type="button" aria-pressed="false" class="rounded-md outline-none text-base px-2 py-1 border box-border text-black bg-neutral-100 hover:bg-neutral-200 dark:bg-neutral-800 dark:text-white dark:border-neutral-700 dark:hover:bg-neutral-800" data-rac="">Toggle Button</button>"`;
